### PR TITLE
feat: support transfering assets to contracts

### PR DESCRIPTION
**Pull request by Torres-ssf**
- Close https://github.com/FuelLabs/fuels-ts/issues/1053

### CODE REVIEW

**Review by arboleya — COMMENTED** [review]
~@Torres-ssf I noticed you added methods to the `@fuel-ts/asm` package and made it a dependency for the `@fuel-ts/providers` and `@fuel-ts/wallet` packages. However, the new methods aren't used anywhere other than the related `opcode.test.ts` file. Am I missing something?~

EDIT: my bad, now I see the `providers` package uses it.

**Review by arboleya — APPROVED** [review]

File: apps/docs/src/guide/wallets/transferring-assets.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
-# Transferring assets
+# Asset Transfers
 
-With `wallet.transfer` you can initiate a transaction to transfer an asset from your wallet to a target address.
+This documentation provides a guide on how to transfer assets between wallets and to contracts using methods in our SDK.
```
Comment (Dhaiwat10) [inline apps/docs/src/guide/wallets/transferring-assets.md:3]:
```suggestion
This documentation provides a guide on how to transfer assets between wallets and to contracts using the SDK.
```
Just a nitpick, for more clarity!

File: apps/docs/src/guide/wallets/transferring-assets.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+
+Here is an illustration on how to use the `wallet.transfer` function:
+
+<<< @/../../docs-snippets/src/guide/wallets/transferring-assets.test.ts#transferring-assets-1{ts:line-numbers}
+
+## Transferring Assets To Contracts
+
+When transferring assets to a deployed contract, we use the `transferToContract` method. This method closely mirrors the `transfer` function in terms of parameter structure.
+
+However, instead of supplying the target wallet's address, as done in `wallet.address` for the transfer method, we input the `id` property of the [Contract](../contracts/) instance representing the target contract.
```
Comment (Dhaiwat10) [inline apps/docs/src/guide/wallets/transferring-assets.md:25]: There might be some cases where someone wants to transfer assets to a contract that they haven't deployed. They won't have access to the `contract` object in this case. We should cover this case too, and instruct them to directly pass in the ID. Thoughts?

**Review by Dhaiwat10 — DISMISSED** [review]
Looking good, just a couple of suggestions:

**Review by arboleya — APPROVED** [review]

**Pull request merged**